packages/drivers/odsp-driver/src/odspDocumentServiceFactoryCore.ts

```diff
@@ -206,15 +211,17 @@ export class OdspDocumentServiceFactoryCore implements IDocumentServiceFactory {
 	 * @param getWebsocketToken - function that can provide a token for accessing the web socket. This is also
 	 * to as the "Push" token in SPO. If undefined then websocket token is expected to be returned with joinSession
 	 * response payload.
-	 * @param storageFetchWrapper - if not provided FetchWrapper will be used
-	 * @param deltasFetchWrapper - if not provided FetchWrapper will be used
 	 * @param persistedCache - PersistedCache provided by host for use in this session.
+	 * @param hostPolicy - Policy for storage provided by host.
+	 * @param snapshotPrefetchResultCache - Non Persistant cache which could contain snapshot promise if it has been
```

Comment:
I'm not sure about this part. I think you are trying to solve more generic problem of - external code controlling lifetime of this cache, and allowing share it across multiple instances of driver factories. I'm not sure consumers do need the latter, and I do not think they would be happy about the former.
I wonder if
1. prefetchLatestSnapshot() should take an instance of the driver factory and leverage that to get access to non-persistent cache
2. consumers need to use same driver factory instance when doing prefetch() calls and loading.

It's hard to say which direction is right, but I think we need to present these two options (your current approach and what I've described) to Loop and figure out which one is easier to work with.

Can you please follow up here?

Reply:
Cosmin preferred to take the approach you suggested, so made changes accordingly.